tango_ros_common/tango_ros_native/include/tango_ros_native/tango_ros_node.h

```diff
@@ -170,13 +173,32 @@ class TangoRosNode {
   double laser_scan_min_height_ = 0;
   double laser_scan_max_height_ = 2.0;
 
-  ros::Publisher fisheye_image_publisher_;
-  sensor_msgs::CompressedImage fisheye_compressed_image_;
-  cv::Mat fisheye_image_;
+  std::shared_ptr<image_transport::ImageTransport> image_transport_;
 
-  ros::Publisher color_image_publisher_;
-  sensor_msgs::CompressedImage color_compressed_image_;
+  image_transport::CameraPublisher fisheye_camera_publisher_;
+  std_msgs::Header fisheye_image_header_;
+  sensor_msgs::CameraInfo fisheye_camera_info_;
+  std::shared_ptr<camera_info_manager::CameraInfoManager> fisheye_camera_info_manager_;
+  image_transport::Publisher fisheye_rectified_image_publisher_;
+  cv::Mat fisheye_image_;
+  cv::Mat fisheye_image_gray_;
```

Comment:
Thanks for sharing the link.
After performing some time measurements, it seems that we don't really save time by having these variables as members from the class. The time is of the same order of magnitude in both cases (~60 ms).
In this case I will remove these variables from the members as suggested by @mcopejans. Same goes for color images variables.